scripts/protocgen.sh

```diff
@@ -1,39 +1,46 @@
 #!/usr/bin/env bash
+set -euo pipefail
 
-set -eo pipefail
+# By default, this script runs against the latest commit to the master branch
+# in the Tendermint spec repository. To use this script with a different version
+# of the spec repository, run it with the $VERS environment variable set to the
+# desired commit in the spec repo.
 
 : ${VERS:=master}
-URL_PATH=archive/
+URL_PATH=tarball/
 if [[ VERS -ne master ]]; then
-    URL_PATH=archive/refs/tags/v
+    URL_PATH=tarball/refs/tags/v
 fi
 
-# Edit this line to clone your branch, if you are modifying protobuf files
-curl -qL "https://github.com/tendermint/spec/${URL_PATH}${VERS}.tar.gz" | tar -xjf - spec-"$VERS"/proto/
+echo "fetching proto files"
 
-cp -r ./spec-"$VERS"/proto/tendermint/** ./proto/tendermint
+# Get shortened ref of commit
+REF=$(curl -H "Accept: application/vnd.github.v3.sha" -qL \
+  "https://api.github.com/repos/tendermint/spec/commits/${VERS}" \
+  | cut -c -7)
+curl -qL "https://api.github.com/repos/tendermint/spec/${URL_PATH}${REF}" | tar -xzf - tendermint-spec-"$REF"/
 
-buf generate --path proto/tendermint
+cp -r ./tendermint-spec-"$REF"/proto/tendermint/* ./proto/tendermint
```

Comment:
Possibly consider:
```suggestion
readonly output="./tendermint-spec-${REF}"
cp -r ${output}/proto/tendermint/* ./proto/tendermint
```

etc.